rtos/TARGET_CORTEX/TARGET_CORTEX_M/mbed_rtx_fault_handler.c

```diff
@@ -196,6 +196,9 @@ void fault_print_str(char *fmtstr, uint32_t *values)
     while(fmtstr[i] != '\0') {
         if(fmtstr[i] == '\n' || fmtstr[i] == '\r') {
             serial_putc(&stdio_uart, '\r');
+            if (fmtstr[i] == '\n') {
+        	    serial_putc(&stdio_uart, '\n');
+            }
```

Comment:
Should this just be:
``` c
if(fmtstr[i] == '\r') {
     serial_putc(&stdio_uart, '\r');
     serial_putc(&stdio_uart, '\n');
```

and let '\r' fall through to the else statement?

Reply:
@geky actually no, this will cause double newline in case of the user will print something like
"hello work double newline \r\n"

Reply:
Oh hmm, won't this turn "\r\n" into "\r\n\r\n" in that case?

I also mistyped in meant
``` c
if(fmtstr[i] == '\r') {
```

In this case "\r\n" would turn into "\r\r\n", but I think the extra "\r" doesn't hurt anything?

Reply:
No, the current implementation will do '\r\r\n'

Reply:
oh I see, gotcha

That would be the same as this right? But with fewer conditionals?
``` c
if(fmtstr[i] == '\n') {
     serial_putc(&stdio_uart, '\r');
     serial_putc(&stdio_uart, '\n');
```

Reply:
not quite as you are discarding any \r char sent to print...
i think the outcome may be the same but in some edge cases it will cause a different issue

Reply:
Hmm, wouldn't the '\r' still go through the else statement and get printed out?
``` c
if(fmtstr[i] == '\n') {
     serial_putc(&stdio_uart, '\r');
     serial_putc(&stdio_uart, '\n');
} else {
     if(fmtstr[i]=='%') {
         hex_to_str(values[vidx++],hex_str);
         for(idx=7; idx>=0; idx--) {
             serial_putc(&stdio_uart, hex_str[idx]);
         }
     } else {
         serial_putc(&stdio_uart, fmtstr[i]);
     }
}
```